mpy-cross/requirements.txt

```diff
@@ -0,0 +1,14 @@
+# wasmtime has the best general compatibility.
+wasmtime==11.0.0
```

Comment:
It would be nice if these were optional dependencies so that we don't always have to install all runtimes, e.g. `pip install mpy-cross[pywasm3]` to get the runtime you actually want to use.

Reply:
I'd like to come up with a requirements.txt that works for as many people as possible. I think what I can do is a combination of "Extras" and "Environment Markers" (following https://peps.python.org/pep-0508/) such that `pip install mpy_cross` "just" works by default, but can be customised with e.g. `pip install mpy-cross[pywasm3]` like you suggest.

At this stage, I think the best option is to come up with an environment marker that will install wasmtime by default for all the supported wasmtime targets, and then the others can be manually chosen.

Reply:
I thought I understood PEP-508 enough, but it seems like I don't.

I've updated the branch with a requirements.txt that will correctly install wasmtime on the platforms that wasmtime supports (64 bit windows, linux, linux-arm, mac). But I don't know how to declare an optional dependency such that e.g. `pip install mpy-cross[pywasm3]` will do what you'd expect (i.e. install the optional dependency with the "pywasm3" extra tag).

@dlech Do you know what I'm missing here? Thanks

Reply:
AFAIK, extras are only additive. So the first thing that comes to mind is that we would need to split this up into an `mpy-cross-core` and an `mpy-cross` package. 

The `mpy-cross` package could include the `__main__.py` and `mpy-cross` script and the "just-works" dependencies. The dependencies could either be `mpy-cross-core[all]` or some platform-specific variations.

The `mpy-cross-core` could contain only the programmatic API and the `mpy-cross` binaries. All dependencies in this package (wasm runtimes) would be under an extra flag, i.e. consumers must either specify an extra dependency or supply their own `mpy-cross` executable.

I suppose we could take this one step further and put the binaries and backend code for each of the various backends in separate `mpy-cross-backend-xyz` packages, but the binaries are probably small enough that it isn't worth the extra effort to go this far.

Reply:
Thinking about it a bit more, having the suggested package split could also allow the `mpy-cross` package to follow the existing versioning scheme of exactly matching the MicroPython version while the `mpy-cross-core` package could actually use proper semver for the Python package itself.